duqtools/config.py

```diff
@@ -60,9 +61,28 @@ class DataLocation(BaseModel):
     run_out_start_at: int
 
 
+class LHSSampler(BaseModel):
+    method: Literal['latin-hypercube']
+    n_samples: int
+
+    def __call__(self, *args):
+        from duqtools.samplers import latin_hypercube
+        return latin_hypercube(*args, n_samples=self.n_samples)
+
+
+class CartesianProduct(BaseModel):
+    method: Literal['cartesian-product'] = 'cartesian-product'
+
+    def __call__(self, *args):
+        from duqtools.samplers import cartesian_product
+        return cartesian_product(*args)
+
+
 class ConfigCreate(BaseModel):
     matrix: List[IDSOperation] = []
-    # template: ImasLocation
+    sampler: Union[LHSSampler,
+                   CartesianProduct] = Field(default=CartesianProduct(),
+                                             discriminator='method')
```

Comment:
This looks quite complex, can't we use a custom validator + getter instead?

Fun thing: as a nasty side effect the validator allowsyou to overload the field with the  function (the return value), see below: but probably a getter would be better.

```python
method_map = { 'cartesian-product' : self.cartesian_product_caller 
                         , 'latin-hypercube' : self.latin_hypercube_caller}

@validator('method')
    def method_nsamples(cls, v, values):
        if not v == 'cartesian-product':
          if not 'n_samples' in values:
            raise ValidationError('n_samples is required if method is not cartesian-product')
        return
```
(the _caller functions should look a lot like the `__call__` function I guess)

But this is also fine the way it is, so feel free to ignore the suggestion

Reply:
Thanks for the suggestion. This is what I was talking about when I was worried when the pydantic config would get a bit complex 😅

Fyi, I took this construction from the `pydantic` docs [here](https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions). What I like about it is that it is shockingly easy to add new samplers (see last commits).

I will check out your suggestion and see what works best 👍